protocols/kad/src/behaviour.rs

```diff
@@ -358,6 +361,7 @@ where
             record_ttl: config.record_ttl,
             provider_record_ttl: config.provider_record_ttl,
             connection_idle_timeout: config.connection_idle_timeout,
+            local_addrs: HashSet::with_capacity(6)
```

Comment:
Both `Addresses` in `addresses.rs` and `local_addrs` use the assumption that a usual node has < 6 addresses in the average case. Do you think this is worth unifying in a single constant?

Reply:
I can add a constant but note that this is really not any kind of important assumption, it is just an arbitrary low constant > 1, since a node rarely has just one listen address. I would even be fine with not using `with_capacity` at all here, given that this is just a single instance per behaviour and we're hence just talking about avoiding (or not) a single allocation per `Kademlia` instance.

Reply:
I am in favor of not using `with_capactiy` here at all.

Reply:
Done in https://github.com/libp2p/rust-libp2p/pull/1708/commits/f40e8a68fd71eb9c38ec017ac33af77199127f4e.